pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+
+        # Exception if property in instance but not class or types do not
+        # match
+        for iprop_name in new_instance:
+            if iprop_name not in target_class.properties:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Property {0!A} specified in NewInstance is not "
+                            "exposed by class {1!A} in namespace {2!A}",
+                            iprop_name, target_class.classname, namespace))
+
+            cprop = target_class.properties[iprop_name]
+            iprop = new_instance.properties[iprop_name]
+            if iprop.is_array != cprop.is_array or \
+                    iprop.type != cprop.type:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Instance and class property {0!A} types "
+                            "do not match: instance={1!A}, class={2!A}",
+                            iprop_name, iprop, cprop))
+
+            # The class and instnames are the same except for possible case
+            # sensitivity. If case different, set cprop_name into new instance
+            # to maintain case equality
+            cprop_name = cprop.name
+            if cprop_name != iprop_name:
+                new_instance.properties[iprop_name].name = cprop_name
+
+        # If property not in instance, add it from class and use default value
+        # from class
+        for cprop_name in target_class.properties:
+            if cprop_name not in new_instance:
+                default_value = target_class.properties[cprop_name]
+                new_instance[cprop_name] = default_value
+
+        # Build instance path. We build the complete instance path
+        new_instance.path = CIMInstanceName.from_instance(
+            target_class,
+            new_instance,
+            namespace=namespace)
+
+        # Check for duplicate instances
+        if instance_store.exists(new_instance.path):
+            raise CIMError(
+                CIM_ERR_ALREADY_EXISTS,
+                _format("NewInstance {0!A} already exists in namespace "
+                        "{1!A}.", new_instance.path, namespace))
+
+        # Reflect the new namespace in the  CIM repository
+        # TODO: This should not be necessary with data store
+        if ns_classname:
+            self.add_namespace(new_namespace)
+
+        # Store the new instance in the  CIM repository
+        instance_store.create(new_instance.path, new_instance)
+
+        # Create instance returns model path, path relative to namespace
+
+        return new_instance.path.copy()
+
+    ####################################################################
+    #
+    #   ModifyInstance server request
+    #
+    ####################################################################
+
+    def ModifyInstance(self, namespace, ModifiedInstance,
+                       IncludeQualifiers=None, PropertyList=None):
```

Comment:
In DSP0200, the INSTANCENAME parameter does not specify a namespace, so it makes sense to have a separate target namespace. However, in this method, the namespace can be specified in the ModifiedInstance parameter and therefore is being checked in the implementation, leading to an exception about mismatching namespaces that would never even exist without that redundancy. I suggest to get rid of the redundancy by removing the `namespace` argument and by requiring that the namespace is set in the ModifiedInstance argument.

Reply:
See comment on previous review item

Reply:
done